### ISSUE TITLE
Minor audio bugs

### ISSUE THREAD

**Issue by coreybruce** [carstene1ns/alex4#3]
Hey first off before I get into the bug report I want to thank you so much for reviving this game and fixing the source code so it can be compiled and played again on Linux natively again as I use to love playing this game on my PC as a kid back in the day on Windows 98! I had a project to revive the game and had a friend help a little to get it to successfully compile than work but we got stuck on getting it to run and my friend didn't have the time to work on it.

The minor audio bugs I wanted to report are that the background music will at a point not loop properly and will just reset to the start instead of looping.

https://user-images.githubusercontent.com/8393780/152671868-adf63e72-8eee-46cb-a943-bfc05ef40129.mp4

Another audio bug is when you finish it will than play the dead music than play the normal game music before cutting to the next level

https://user-images.githubusercontent.com/8393780/152671971-7b6f6ad7-29b3-4ba6-9767-6e50cb9830b8.mp4

**Comment by carstene1ns** [comment]
This both looks like a bug in looping/time keeping. The music and jingles are concatenated in the mod file. Maybe also a bug in dumb library. Need to investigate.

**Comment by absolutelynothinghere** [comment]
>Maybe also a bug in dumb library.

I thought this port didn't use libdumb? I compiled the game with SDL2_mixer and libmodplug.

**Comment by carstene1ns** [comment]
Well, this one is related to the old allegro version of the game, I started the SDL conversion after this issue was opened. But since I have not released the SDL port formally (as 1.3), this is still relevant to the allegro version.

**Comment by absolutelynothinghere** [comment]
Actually the SDL port suffers from audiobug1, so this might be an issue with alex4 rather than libmodplug/SDL2_mixer. I haven't encountered audiobug2 though thankfully.

P.S. considering the SDL port introduces significant changes (addition of color mode and removal of level editor) it might be better to name the upcoming version 2.0.0. This is in line with [semantic versioning](https://semver.org).

**Comment by coreybruce** [comment]
> Actually the SDL port suffers from audiobug1, so this might be an issue with alex4 rather than libmodplug/SDL2_mixer. I haven't encountered audiobug2 though thankfully.
> 
> P.S. considering the SDL port introduces significant changes (addition of color mode and removal of level editor) it might be better to name the upcoming version 2.0.0. This is in line with [semantic versioning](https://semver.org).

Would it be better to jump to SDL3? also would the editor be readded?

**Comment by carstene1ns** [comment]
The editor and the shooter minigame are currently disabled, since they require more work yet (see also [TODO.md](https://github.com/carstene1ns/alex4/blob/master/TODO.md)).
While the editor works mostly, I have not touched the shooter game yet.

Fixing the audio bugs is still on my list.

Porting to SDL3 should be easy, but it needs adoption in the major linux distributions yet.
Also SDL2 should be supported for the near future (considering SDL1.2 is still available nowadays and is more than 20 years old).

**Comment by coreybruce** [comment]
> The editor and the shooter minigame are currently disabled, since they require more work yet (see also [TODO.md](https://github.com/carstene1ns/alex4/blob/master/TODO.md)). While the editor works mostly, I have not touched the shooter game yet.
> 
> Fixing the audio bugs is still on my list.
> 
> Porting to SDL3 should be easy, but it needs adoption in the major linux distributions yet. Also SDL2 should be supported for the near future (considering SDL1.2 is still available nowadays and is more than 20 years old).

yeah might be better to migrate to SDL2 then SDL3 later once it's in major Linux distros
